pypesto/profile/walk_along_profile.py

```diff
@@ -78,26 +79,168 @@ def walk_along_profile(
+            # Check max_step_size is not reduced below min_step_size
+            if (
+                options.max_step_size * max_step_reduce_factor
+                < options.min_step_size
+            ):
+                logger.warning(
+                    "Max step size reduced below min step size. "
+                    "Setting a lower min step size can help avoid this issue."
+                )
+                break
+
+            # compute the new start point for optimization
+            x_next = create_next_guess(
+                x_now,
+                i_par,
+                par_direction,
+                options,
+                current_profile,
+                problem,
+                global_opt,
+                1.0,
+                max_step_reduce_factor,
+            )
+
+            # fix current profiling parameter to current value and set start point
+            problem.fix_parameters(i_par, x_next[i_par])
+            startpoint = x_next[problem.x_free_indices]
+
+            if startpoint.size > 0:
+                optimizer_result = optimizer.minimize(
+                    problem=problem,
+                    x0=startpoint,
+                    id=str(0),
+                    optimize_options=OptimizeOptions(
+                        allow_failed_starts=False
+                    ),
+                )
+
+                if np.isfinite(optimizer_result.fval):
+                    optimization_successful = True
+                    if max_step_reduce_factor == 1.0:
+                        # The color of the point is set to black if no changes were made
+                        color_next = np.array([0, 0, 0, 1])
+                    else:
+                        # The color of the point is set to red if the max_step_size was reduced
+                        color_next = np.array([1, 0, 0, 1])
+                else:
+                    max_step_reduce_factor *= 0.5
+                    logger.warning(
+                        f"Optimization at {problem.x_names[i_par]}={x_next[i_par]} failed. "
+                        f"Reducing max_step_size to {options.max_step_size * max_step_reduce_factor}."
+                    )
+            else:
+                # if too many parameters are fixed, there is nothing to do ...
+                fval = problem.objective(np.array([]))
+                optimizer_result = OptimizerResult(
+                    id="0",
+                    x=np.array([]),
+                    fval=fval,
+                    n_fval=0,
+                    n_grad=0,
+                    n_res=0,
+                    n_hess=0,
+                    n_sres=0,
+                    x0=np.array([]),
+                    fval0=fval,
+                    time=0,
+                )
+                optimizer_result.update_to_full(problem=problem)
+                optimization_successful = True
+                color_next = np.concatenate((color_now[:3], [0.3]))
+
+        if not optimization_successful:
```

Comment:
this is a lot of intendation. It might be savable with a `if optimization_sucessful: current_profile.append... return current profile`?

Reply:
with below comment this might not be necessary